lib/kitchen/driver/oci.rb

```diff
@@ -269,29 +271,27 @@ def inject_powershell(state)
         }
       end
 
-      def base_oci_launch_details
-        request = OCI::Core::Models::LaunchInstanceDetails.new
+      def generate_hostname
         prefix = config[:hostname_prefix]
-        if prefix.empty?
-          hostname = random_hostname(instance.name)
-        else
-          hostname = random_hostname(prefix + '-' + instance.name)
-        end
+        [prefix, random_hostname(instance.name)].compact.join('-')
+      end
```

Comment:
` useful method if you need to remove nil values from an array is compact` - hey, thats neat!

Reply:
We'll make a Rubyist out of you yet :)